pkg/observer/controller.go

```diff
@@ -309,6 +370,15 @@ func (c *controller) Run() {
 	klog.V(3).Infof("waiting for %v to allow changes to settle", c.WaitInterval)
 	select {
 	case <-time.After(c.WaitInterval):
+		timeout := 0
+		retries := 20
+		for c.safeToStartCycle() == false {
```

Comment:
This new logic of trying to apply the CNRs should be in a seperate function - as having another set of sleeping/waiting inside a select starts getting complicated.

Reply:
Ok, I can pull it out